crates/db_connection_pool/src/snowflakepool.rs

```diff
@@ -81,19 +78,21 @@ impl SnowflakeConnectionPool {
             .context(MissingRequiredSecretSnafu { name: "username" })?;
         let password = get_param(params, secret, "password")
             .context(MissingRequiredSecretSnafu { name: "password" })?;
-
         let account = get_param(params, secret, "account")
-            .context(MissingRequiredParameterSnafu { name: "account" })?;
-        let warehouse = get_param(params, secret, "warehouse")
-            .context(MissingRequiredParameterSnafu { name: "warehouse" })?;
+            .context(MissingRequiredSecretSnafu { name: "account" })?;
+        let warehouse = get_param(params, secret, "snowflake_warehouse");
+        let role = get_param(params, secret, "snowflake_role");
+
+        // account identifier can be in <orgname.account_name> format but API requires it as <orgname-account_name>
+        let account = account.replace('.', "-");
```

Comment:
Can org/account names have periods `.`?

Reply:
Confirm both org name and user name can't contain '.' in name
![image](https://github.com/spiceai/spiceai/assets/981580/4d76c587-5e46-4749-9eeb-3312760c07b8)